commit 2c5209ac671bd4bbe3c00d6e81df345c93999fda
Author: Andy Whitcroft <apw@shadowen.org>
Date:   Wed Feb 15 08:47:05 2012 -0700

    UBUNTU: SAUCE: allow IRQs to be irq-threaded by default via config
    
    Signed-off-by: Andy Whitcroft <apw@canonical.com>

diff --git a/kernel/irq/Kconfig b/kernel/irq/Kconfig
index a117adf7084b..c189143df6cd 100644
--- a/kernel/irq/Kconfig
+++ b/kernel/irq/Kconfig
@@ -111,6 +111,10 @@ config IRQ_DOMAIN_DEBUG
 config IRQ_FORCED_THREADING
        bool
 
+config IRQ_FORCED_THREADING_DEFAULT
+	bool "Use IRQ threading by default"
+	depends on IRQ_FORCED_THREADING
+
 config SPARSE_IRQ
 	bool "Support sparse irq numbering" if MAY_HAVE_SPARSE_IRQ
 	---help---
diff --git a/kernel/irq/manage.c b/kernel/irq/manage.c
index 4bff6a10ae8e..aeae7662a3ce 100644
--- a/kernel/irq/manage.c
+++ b/kernel/irq/manage.c
@@ -24,14 +24,20 @@
 #include "internals.h"
 
 #ifdef CONFIG_IRQ_FORCED_THREADING
-__read_mostly bool force_irqthreads;
+__read_mostly bool force_irqthreads = IS_ENABLED(CONFIG_IRQ_FORCED_THREADING_DEFAULT);
 
 static int __init setup_forced_irqthreads(char *arg)
 {
 	force_irqthreads = true;
 	return 0;
 }
+static int __init setup_no_irqthreads(char *arg)
+{
+	force_irqthreads = false;
+	return 0;
+}
 early_param("threadirqs", setup_forced_irqthreads);
+early_param("nothreadirqs", setup_no_irqthreads);
 #endif
 
 static void __synchronize_hardirq(struct irq_desc *desc)
